operator/Dockerfile

```diff
@@ -16,11 +16,11 @@ COPY controllers/ controllers/
 COPY internal/ internal/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -mod=readonly -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=readonly -a -o manager main.go
```

Comment:
We use this Dockerfile to build multi-arch production images on [docker.io/grafana/loki-operator](https://hub.docker.com/r/grafana/loki-operator). So we cannot simply pin to `amd64`

Reply:
Gotcha. Hmm, maybe there should be a note in the developer notes for this then. As is, the `Dockerfile` does not work on an M1 Mac.